### PR TITLE
Add local br e2e test

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Add e2e case for Backup and Restore by BR, store data in minio.

close https://github.com/pingcap/tidb-operator/issues/2347

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -287,14 +284,50 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		oa.StopInsertDataTo(&clusterA)
 	})
 
-	ginkgo.It("Adhoc backup and restore with BR CRD", func() {
-		if framework.TestContext.Provider != "aws" {
-			framework.Skipf("provider is not aws, skipping")
+	ginkgo.It("BackupAndRestoreWithBR", func() {
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go:287]:
```suggestion
	ginkgo.It("Backup and restore with BR", func() {
```

describe in sentence form

File: tests/e2e/tidbcluster/storage.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+const (
+	minioPodName = "minio"
+)
+
+type storage interface {
```
Comment (cofyc) [inline tests/e2e/tidbcluster/storage.go:40]: move this `storage` interface and implementations to a separate package, e.g. `tests/e2e/util/storage`? I think it deserves a separate package.

**Comment by cofyc** [comment]
job on aws: https://internal.pingcap.net/idc-jenkins/view/tidb-operator/job/tidb-operator-e2e-eks-ci-master/92/console

**Comment by cofyc** [comment]
cc @shuijing198799 PTAL

**Comment by Yisaer** [comment]
@cofyc  It seems the test in AWS has been succeeded as I found the log `20:44:09  [2] May  9 12:44:09.122: INFO: clean backup success`

File: tests/e2e/tidbcluster/storage.go (hunk 2 above)
Comment (Yisaer) [reply tests/e2e/tidbcluster/storage.go]: updated.

**Pull request merged**